torch/_inductor/lowering.py

```diff
@@ -3814,12 +3815,91 @@ def loader(idx):
     )
 
 
+def inplace_constant_pad_nd(x, padding, fill_value):
+    """
+    This optimization changes the semantics of padding from 'clone'
+    style to 'view' style.
+
+    Thanks to functionalization, this change can still maintain numerical
+    correctness.
+    """
+
+    def _padding_can_be_fused():
+        """
+        Conservatively check if padding can be fused with downstream op.
+        1. if the downstream op is a sum, then there is little benefit to
+           do inplace padding
+        2. if the downstream op is a matmul, doing inplace padding can
+           save membw.
+        """
+        current_node = V.graph.current_node
+        if current_node is None:
+            return True  # be conservative
+        users = tuple(current_node.users)
+        if len(users) == 1 and users[0].target in (
+            aten.mm.default,
+            aten.addmm.default,
+        ):
+            return False
+
+        return True  # be conservative
+
+    if _padding_can_be_fused():
+        return None
+
+    # Only handle 2D case for now
+    if len(padding) != 4 or len(x.get_size()) != 2:
+        return None
+
+    if not ir.is_storage_and_layout(x):
```

Comment:
I'm not sure actually. This function is written in a way that we don't slow things down if we can not do the inplace-padding.  I'm worried if 'realize' here will slow down some cases? Or at lease increase the number of SchedulerNode even if it does not impact perf since we can fuse in scheduler anyways.

Reply:
At this point we've already checked that it has a single, unfusible use so I dont think so.. but potentially wed want to freeze to a non-dense layout so that we could do the optimization.. in any case should be fine as is for now

Reply:
yea, good point